src/project/mod.rs

```diff
@@ -0,0 +1,60 @@
+//! Project management module
+//!
+//! Manages configuration files, paths, and such.
+
+use std::io;
+use std::path::{Path, PathBuf};
+
+/// Configuration macros.
+#[macro_use]
+pub mod config;
+/// Directory and file loading.
+pub mod directory;
+
+pub use self::config::Config;
+
+/// Error related to anything that manages/creates configurations as well as
+/// "workspace"-related things.
+#[derive(Debug)]
+pub enum ProjectError {
+    /// Forward to the `std::io::Error` error.
+    File(io::Error),
+    /// Errors related to serde's parsing of configuration files.
+    Parser(String),
+}
+
+impl From<io::Error> for ProjectError {
+    fn from(e: io::Error) -> ProjectError {
+        ProjectError::File(e)
+    }
+}
+
+impl ProjectError {
+    /// Returns a human friendly error message for the `ProjectError`.
+    pub fn description(&self) -> String {
+        match self {
+            &ProjectError::File(ref err) => err.to_string(),
+            &ProjectError::Parser(ref msg) => msg.clone(),
+        }
+    }
+}
+
+/// Project structure, holds information related to configurations and meta information.
+pub struct Project {
```

Comment:
Probably missing a `#[derive(Debug)]`? Stupid question: Sorry if I missed some discussion about this, but how is this supposed to be used? Also, why isn't it using `Directory`? I think it would make sense to have it load all the configs together.